src/image_gen/imagegenutils.cpp

```diff
@@ -381,44 +444,94 @@ std::variant<absl::Status, ov::AnyMap> getImageEditRequestOptions(const ovms::Ht
-    // quality string REJECTED  NO optional default=auto
-    // response_format string NO optional default=url
-    // size DONE optional default=auto
-    // user string REJECTED optional default=vivid
+    // now get optional string parameters
+    for (auto key : {"prompt_2", "prompt_3", "negative_prompt", "negative_prompt_2", "negative_prompt_3"}) {
+        SET_OPTIONAL_KEY_OR_RETURN(std::string, getStringFromPayload);
+    }
+    SET_OR_RETURN(std::optional<std::string>, responseFormatOpt, getStringFromPayload(payload, "response_format"));
+    if (responseFormatOpt.has_value() && responseFormatOpt.value() != "b64_json") {
+        return absl::InvalidArgumentError(absl::StrCat("Unsupported response_format: ", responseFormatOpt.value(), ". Only b64_json is supported."));
+    }
+    // now get optional int parameters
+    SET_OR_RETURN(std::optional<int>, nOpt, getIntFromPayload(payload, "n"));
+    INSERT_IF_HAS_VALUE_RETURN_IF_FAIL("num_images_per_prompt", nOpt);
+    for (auto key : {"num_images_per_prompt", "max_sequence_length"}) {
+        SET_OPTIONAL_KEY_OR_RETURN(int, getIntFromPayload);
+    }
+    // now get optional float parameters
+    for (auto key : {"guidance_scale", "strength"}) {
+        SET_OPTIONAL_KEY_OR_RETURN(float, getFloatFromPayload);
+    }
+    // now get optional int64_t parameters
+    for (auto key : {"width", "height"}) {
+        SET_OPTIONAL_KEY_OR_RETURN(int64_t, getInt64FromPayload);
+    }
+    // now get optional size_t parameters
+    for (auto key : {"num_inference_steps", "rng_seed"}) {
+        SET_OPTIONAL_KEY_OR_RETURN(size_t, getSizetFromPayload);
+    }
+    // return error on unhandled parameters
+    // background/moderation/output_compression/output_format/quality/style
+    // TODO possibly to be handled outside since output_compresiont/format are nonGenai
+    for (auto key : {"background", "moderation", "output_compression", "output_format", "quality", "style"}) {
+        auto value = payload.getFieldByName(key);
+        if (!value.empty()) {
+            return absl::InvalidArgumentError(absl::StrCat("Unhandled parameter: ", key));
+        }
+    }
+    // now insert default values if not already populated ?
+    if (args.defaultResolution.has_value()) {
+        if (requestOptions.find("height") == requestOptions.end()) {
+            requestOptions.insert({"height", args.defaultResolution->second});
+        }
+        if (requestOptions.find("width") == requestOptions.end()) {
+            requestOptions.insert({"width", args.defaultResolution->first});
+        }
+    }
+    // now check if in httpPaylod.parsedJson we have any fields other than the ones we accept
+    // accepted are: prompt, prompt_2, prompt_3, negative_prompt, negative_prompt_2, negative_prompt_3,
+    // num_images_per_prompt, max_sequence_length, height, width, rng_seed, num_inference_steps,
+    // strength, guidance_scale, size
+    // if we have any other fields return error
```

Comment:
Should we do that? We have experiences when to strict validation lead to non functional external samples when API is extended. Wouldn't it be safer to just ignore it?

Reply:
@atobiszei decision, and i think it is good to coordinate all changes with GenAI, or we will simply introduce unexpected behavior when something is added without verification